tools/clang_format_wrapper.py

```diff
@@ -0,0 +1,143 @@
+    RENAMED = "R"
+    UNMODIFIED = " "
+    UNTRACKED = "?"
+    UPDATED_BUT_UNMERGED = "U"
+
+
+GitEdit = namedtuple("GitEdit", ["x", "y", "path"])
+
+
+def added_and_modified_paths() -> Iterable[str]:
+    """Inspect `git status` output.
+
+    Return paths to be modified by `git-clang-format`.
+    """
+
+    command = ["git", "status", "--porcelain", "--untracked-files=no"]
+    logger.debug("Executing `%s`", " ".join(command))
+    git_status_output = check_output(args=command, text=True)
+
+    edits = [
+        edit for edit in git_status_output.rstrip().split("\n") if edit != ""
+    ]
+    logger.debug(
+        "Found %d path(s) with state change in GIT status output", len(edits)
+    )
+
+    if edits:
+        logger.debug("Pending GIT edits:\n%s", "\n".join(edits))
+
+    return (
+        git_edit.path
+        for git_edit in (
+            GitEdit(edit[0:1], edit[1:2], edit[3:]) for edit in edits
+        )
+        if git_edit.x in (GitState.MODIFIED.value, GitState.ADDED.value)
+    )
+
+
+def run_clang_format(file_list: List[str]) -> bool:
+    """Execute git-clang-format for provided `file_list`.
+
+    Return True if `git-clang-format` has errors or has modified any file,
+    i.e. if it failed the formatting test.
+    """
+    if not file_list:
+        logger.info("No file to be formatted, exiting")
+        return
+
+    command = ["git-clang-format"]
+
+    src_commit = env.get("PRE_COMMIT_FROM_REF") or env.get("PRE_COMMIT_SOURCE")
+
+    if src_commit:
+        command.append(src_commit)
+
+    command.append("--")
+    command.extend(file_list)
+
+    try:
+        logger.info("Executing `%s`", " ".join(command))
+        output = check_output(command, text=True)
+        logger.debug(output)
+
+        if output == "clang-format did not modify any files\n":
+            return False
+
+    except CalledProcessError as error:
+
+        if error.returncode == 2:
+            # Handled case: unstaged changes
+            return True
+        else:
+            raise error
+
+    return True
+
+
+def main():
+    # Setup CLI argument parser
+    logging.basicConfig(
+        format="[%(levelname)s] %(message)s", level=logging.DEBUG
+    )
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "files", nargs="*", help="Restrict formatting to these files"
+    )
+    args = parser.parse_args()
+    logger.debug("Wrapper called with `files` arguments: %s", args.files)
+
+    # Inspect GIT status output and filter paths that must be clang-formatted
+    paths_to_be_clang_formatted = [
+        path for path in added_and_modified_paths() if path in args.files
+    ]
+    logger.debug(
+        "Found %d staged path(s) to be clang-formatted",
+        len(paths_to_be_clang_formatted),
+    )
+
+    if paths_to_be_clang_formatted:
+        logger.debug("Paths:\n%s", "\n".join(paths_to_be_clang_formatted))
+
+        if run_clang_format(paths_to_be_clang_formatted):
+            sys.exit(1)
```

Comment:
If you simply `return` the exit code from `main` and call `sys.exit` in line 143, you can actually import and run this function from another script without needing to catch `SystemExit` or risking that it exits your outer script.